charms/kfp-persistence/src/charm.py

```diff
@@ -39,6 +41,8 @@ def __init__(self, *args):
         self.framework.observe(self.on.install, self.set_pod_spec)
         self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
+        # TODO: Need to add a relation_changed hook here too, otherwise
```

Comment:
Isn't this already happening underneath?

I think we should just update the status handling to our [new way of doing things](https://github.com/canonical/kubeflow-profiles-operator/pull/18)